migrations/blocked_tasks.js

```diff
@@ -0,0 +1,83 @@
+// constants
+finishedStatuses = ["success","failed"]
+
+function taskIsBlocked(task) {
+    for(dep of task.depends_on) {
+        if (dep.unattainable) {
+            return true
+        }
+    }
+
+    return false
+}
+
+function unattainableResolved(task) {
+    for (dep of task.depends_on) {
+        if (dep.unattainable == null) {
+            return false
+        }
+    }
+    return true
+}
+
+//
+// example invocation:
+//
+// startingDate = ISODate("2016-01-01T00:00:00Z")
+// millisecondsPerDay = 86400000
+// delta = millisecondsPerDay * 1
+// sleepMilliseconds = 100
+// migrate(startingDate, delta, sleepMilliseconds)
+
+function migrate(startingDate, delta, sleepMilliseconds) {
+    while(startingDate < new Date()) {
+        loops = 0
+        tasksSize = 0
+        while (true) {
+            tasks = db.tasks.find({"create_time": {"$gte": startingDate, "$lte": new Date(startingDate.getTime() + delta)}, "status":"undispatched", "depends_on.status": {"$in": ["success", "failed", "", "*"]}, "depends_on": {"$elemMatch":{"unattainable": {"$exists": false}}}}, {"depends_on":1}).toArray()
+            printjson(loops++)
```

Comment:
maybe put the starting date, and log after line 43

Reply:
I moved the logging of loop count.
I put logging the startDate in the beginning of the outer loop.